### ISSUE TITLE
Profile Pictures can not be edited on Huawei device 

### ISSUE THREAD

**Issue by riyapenn-signal** [signalapp/Signal-Android#7726]
- [X] I have searched open and closed issues for duplicates. Same as https://github.com/signalapp/Signal-Android/issues/7340 but with updated information and logs.
- [X] I am submitting a bug report for existing functionality that does not work as intended
- [X] I have read https://github.com/signalapp/Signal-Android/wiki/Submitting-useful-bug-reports
- [X] This isn't a feature request or a discussion topic

----------------------------------------

### Bug description
Can not edit a profile picture on a Huawei phone.

### Steps to reproduce
- Navigate to Signal Settings
- Tap on the profile avatar

**Actual result:** Signal shows a white/blank screen, and returns to the settings options. The profile picture does not change
**Expected result:** See the screen to update the profile.

### Screenshots
N/A

### Device info

**Device:** HUAWEI HUAWEI VNS-L31
**Android version:** 7.0 / EMUI 5.0.1
**Signal version:** 4.18.3

### Link to debug log
 https://debuglogs.org/1db05d5c79de9ef5a86fc6b76655275a2b7fd5af8ee56686a0fc1739967bb5b4 

```

--------- beginning of crash
  | 04-24 00:37:41.786  9739  9739 E AndroidRuntime: FATAL EXCEPTION: main
  | 04-24 00:37:41.786  9739  9739 E AndroidRuntime: Process: org.thoughtcrime.securesms, PID: 9739
  | 04-24 00:37:41.786  9739  9739 E AndroidRuntime: java.lang.IllegalArgumentException: Failed to find configured root that contains /storage/3331-3939/Android/data/org.thoughtcrime.securesms/cache/capture529450042jpg
  | 04-24 00:37:41.786  9739  9739 E AndroidRuntime: 	at android.support.v4.content.FileProvider$SimplePathStrategy.getUriForFile(FileProvider.java:719)
  | 04-24 00:37:41.786  9739  9739 E AndroidRuntime: 	at android.support.v4.content.FileProvider.getUriForFile(FileProvider.java:404)
  | 04-24 00:37:41.786  9739  9739 E AndroidRuntime: 	at org.thoughtcrime.securesms.util.FileProviderUtil.getUriFor(FileProviderUtil.java:17)
  | 04-24 00:37:41.786  9739  9739 E AndroidRuntime: 	at org.thoughtcrime.securesms.CreateProfileActivity.createAvatarSelectionIntent(CreateProfileActivity.java:381)
  | 04-24 00:37:41.786  9739  9739 E AndroidRuntime: 	at org.thoughtcrime.securesms.CreateProfileActivity.handleAvatarSelectionWithPermissions(CreateProfileActivity.java:412)
  | 04-24 00:37:41.786  9739  9739 E AndroidRuntime: 	at org.thoughtcrime.securesms.CreateProfileActivity.bridge$lambda$0$CreateProfileActivity(CreateProfileActivity.java)
  | 04-24 00:37:41.786  9739  9739 E AndroidRuntime: 	at org.thoughtcrime.securesms.CreateProfileActivity$$Lambda$7.run(Unknown Source)
  | 04-24 00:37:41.786  9739  9739 E AndroidRuntime: 	at org.thoughtcrime.securesms.permissions.PermissionsRequest.onResult(PermissionsRequest.java:85)
  | 04-24 00:37:41.786  9739  9739 E AndroidRuntime: 	at org.thoughtcrime.securesms.permissions.Permissions$PermissionsBuilder.executePreGrantedPermissionsRequest(Permissions.java:152)
  | 04-24 00:37:41.786  9739  9739 E AndroidRuntime: 	at org.thoughtcrime.securesms.permissions.Permissions$PermissionsBuilder.execute(Permissions.java:140)
  | 04-24 00:37:41.786  9739  9739 E AndroidRuntime: 	at org.thoughtcrime.securesms.CreateProfileActivity.lambda$initializeResources$0$CreateProfileActivity(CreateProfileActivity.java:225)
  | 04-24 00:37:41.786  9739  9739 E AndroidRuntime: 	at org.thoughtcrime.securesms.CreateProfileActivity$$Lambda$0.onClick(Unknown Source)
  | 04-24 00:37:41.786  9739  9739 E AndroidRuntime: 	at android.view.View.performClick(View.java:5646)
  | 04-24 00:37:41.786  9739  9739 E AndroidRuntime: 	at android.view.View$PerformClick.run(View.java:22473)
  | 04-24 00:37:41.786  9739  9739 E AndroidRuntime: 	at android.os.Handler.handleCallback(Handler.java:761)
  | 04-24 00:37:41.786  9739  9739 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:98)
  | 04-24 00:37:41.786  9739  9739 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:156)
  | 04-24 00:37:41.786  9739  9739 E AndroidRuntime: 	at android.app.ActivityThread.main(ActivityThread.java:6523)
  | 04-24 00:37:41.786  9739  9739 E AndroidRuntime: 	at java.lang.reflect.Method.invoke(Native Method)
  | 04-24 00:37:41.786  9739  9739 E AndroidRuntime: 	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:942)
  | 04-24 00:37:41.786  9739  9739 E AndroidRuntime: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:832)
```

**Comment by jlaufs** [comment]
I experience the exact same problem. See my debug log here: https://github.com/signalapp/Signal-Android/issues/7766

**Comment by jlaufs** [comment]
I experience the exact same problem. See my debug log here: 

https://debuglogs.org/d7c45197615458c36e0a777a16b05cc093e2d16eb99363334b1b2d137b1bf8d7

**Comment by eppfel** [comment]
Same problem on Huawei P9 Lite.

Seems the way Huawei handles storage is a bit weird. I have set the SD card as default storage.

Device  : HUAWEI HUAWEI VNS-L31 (VNS-L31)
Android : 7.0 (C432B412, VNS-L31C432B412)
Memory  : 4M (18.24% free, 512M max)
Memclass: 384
OS Host : WUH1000129219
App     : Signal 4.24.5

**Comment by florianschroen** [comment]
Same on (Huawei) Honor 7x.
SD card is configured as default storage. 

Device  : HUAWEI BND-L21 (BND-L21)
Android : 8.0.0 (330(C432), BND-L21 8.0.0.330(C432))
Memory  : 10M (8.66% free, 512M max)
Memclass: 384
OS Host : WUH1000128887
App     : Signal 4.24.8

**Comment by greyson-signal** [comment]
This will be fixed in 4.26.

**Comment by nmvega** [comment]
As of `v4.30.8` this is still an issue (not fixed). The only options for changing the image are `delete`, `camera` (which I don't want) and `icon programs`. There is no option to browse from a file manager and select an image; no gallery option; etc. Therefore I can't set it. Bug.
